### PR TITLE
Remove some cli options. Resolves #1857

### LINKED ISSUE #1857: Consolidate command line options
As I mentioned in #1848, here are some options for consolidating the command line options:

*    `--local-time` should be a minor option at most (perhaps only configurable through config.rb)?
*   `--lenient-tag-lookup` could be removed in 6.x, of course saving the actual compatibility feature and documenting how to enable it through config.rb.
*    `--default-keybinding` could potentially be removed and again made into config.rb option (but needn't be)

### PR DESCRIPTION
**Pull request by dometto**
@bartkamphorst can I have a sanity check on this? Is cleaning up the list of CLI options a little bit worth 'obscuring' the removed options?

### CODE REVIEW

**Comment by benjaminwil** [comment]
I think this change makes sense. It could be worthwhile to ensure that we have explicit update documentation for users using these options. Some users may have to go from not using a config file to using one if they needed to preserve these settings.

File: README.md [hunk 1]
```diff
@@ -133,8 +133,6 @@ Gollum comes with the following command line options:
 | --css             | none      | Tell Gollum to inject custom CSS into each page. Uses `custom.css` from wiki root.<sup>3</sup> |
 | --js              | none      | Tell Gollum to inject custom JS into each page. Uses `custom.js` from wiki root.<sup>3</sup> |
 | --no-edit         | none      | Disable the feature of editing pages. |
-| --local-time | none      | Use the browser's local timezone instead of the server's for displaying dates. Default: false.
-| --follow-renames, --no-follow-renames  | none      | Follow pages across renames in the History view. Default: true.
 | --allow-uploads   | [MODE]    | Enable file uploads. If set to `dir`, Gollum will store all uploads in the `/uploads/` directory in repository root. If set to `page`, Gollum will store each upload at the currently edited page.<sup>4</sup> |
 | --mathjax         | none      | Enables MathJax (renders mathematical equations). By default, uses the `TeX-AMS-MML_HTMLorMML` config with the `autoload-all` extension.<sup>5</sup> |
```
Comment (bartkamphorst) [inline README.md:137]: This `--mathjax` option has changed as well, right?

File: README.md [hunk 2]
```diff
@@ -146,7 +144,6 @@ Gollum comes with the following command line options:
 | --template-dir    | [PATH]    | Specify custom mustache template directory. Only overrides templates that exist in this directory. |
 | --template-page   | none      | Use _Template in root as a template for new pages. Must be committed. |
 | --emoji           | none      | Parse and interpret emoji tags (e.g. `:heart:`) except when the leading colon is backslashed (e.g. `\:heart:`). |
```
Comment (bartkamphorst) [inline README.md:146]: Should we add the default of this option to the explanation?

**Review by bartkamphorst — APPROVED** [review]
The `--mathjax` option has changed as well, right?

**Comment by dometto** [comment]
@benjaminwil this is merged and the [release notes](https://github.com/gollum/gollum/wiki/6.0-Release-Notes) seem up to date so that folks will now how to migrate. Do you want to do the honours and release 6.0?

**Comment by benjaminwil** [comment]
6.0 has been released!

**Pull request merged**